test/units/plugins/strategy/test_notify.py

```diff
@@ -0,0 +1,238 @@
+        """,
+    })
+
+    playbook_file = setup_test_env(test_case)
+    handlers, dummy = run_playbook(playbook_file, strategy_class, monkeypatch)
+    assert handlers == ['with_handler : myhandler']
+
+
+@pytest.mark.parametrize('strategy_class', [LSM, FSM])
+def test_include_role_loop_listen(setup_test_env, strategy_class, monkeypatch):
+    test_case = base_test_case.copy()
+    test_case.update({
+        'playbook.yml': """
+        - hosts: localhost
+          gather_facts: no
+          tasks:
+          - include_role: name=with_handler
+            loop: [1, 2, 3]
+        """,
+        'roles/with_handler/handlers/main.yml': """
+        - name: first
+          debug: msg=myhandler
+          listen: myhandler
+        - name: second
+          debug: msg=myhandler
+          listen: myhandler
+        """,
+    })
+
+    playbook_file = setup_test_env(test_case)
+    handlers, dummy = run_playbook(playbook_file, strategy_class, monkeypatch)
+    assert handlers == ['with_handler : first', 'with_handler : second']
+
+
+@pytest.mark.parametrize('strategy_class', [LSM, FSM])
+def test_notify_in_handler(setup_test_env, strategy_class, monkeypatch):
+    test_case = base_test_case.copy()
+    test_case.update({
+        'roles/with_handler/handlers/main.yml': """
+        - name: myhandler
+          command: echo a
+          notify: myhandler2
+        - name: myhandler2
+          command: echo a
+        """,
+    })
+
+    playbook_file = setup_test_env(test_case)
+    handlers, dummy = run_playbook(playbook_file, strategy_class, monkeypatch)
+    assert handlers == ['with_handler : myhandler', 'with_handler : myhandler2']
+
+
+@pytest.mark.parametrize('strategy_class', [LSM, FSM])
+def test_include_task_in_handler(setup_test_env, strategy_class, monkeypatch):
+    test_case = base_test_case.copy()
+    test_case.update({
+        'roles/with_handler/tasks/my_tasks.yml': """
+        - name: role task
+          command: echo a
+        """,
+        'roles/with_handler/handlers/main.yml': """
+        - name: myhandler
+          include_tasks: my_tasks.yml
+        """,
+    })
+
+    playbook_file = setup_test_env(test_case)
+    handlers, dummy = run_playbook(playbook_file, strategy_class, monkeypatch)
+    assert handlers == ['with_handler : myhandler', 'with_handler : role task']
+
+
+@pytest.mark.parametrize('strategy_class', [LSM, FSM])
+def test_first_dup_win(setup_test_env, strategy_class, monkeypatch):
+    test_case = base_test_case.copy()
+    test_case.update({
+        'roles/with_handler/handlers/main.yml': """
+        - name: myhandler
+          debug:
+            msg: first win
+        - name: myhandler
+          debug:
+            msg: last is ignored
+        """,
+    })
+
+    playbook_file = setup_test_env(test_case)
+    handlers, results = run_playbook(playbook_file, strategy_class, monkeypatch)
+    assert handlers == ['with_handler : myhandler']
+    assert results[-1]['msg'] == 'first win'
```

Comment:
This is inconsistent, in general we have "last loaded wins" but in role's handlers we do the opposite?

Reply:
@mkczuk , my patch didn't change anything. This test passes with the devel branch and I just reproduced the behaviour with the v2.7.9 version.

Reply:
I know, was just wondering if it's intended that it works that way.

Reply:
last loaded wins, but iirc, we reverse search on handlers list

Reply:
So:

- if I've several handler files, the last comes first.
- in an handler file, the handler comes first.

I just added this test. The "listen" handlers come first. 
https://github.com/ansible/ansible/pull/53644/commits/354c38a0fbae22db7b90abb95970c76409b49403